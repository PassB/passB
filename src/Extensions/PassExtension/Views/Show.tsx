import {LazyInject} from "Decorators/LazyInject";
import {PassCli} from "PassCli";
import * as React from 'react';
import {RouteComponentProps} from "react-router";

interface LocationStateProps {
  location: {
    state: {
      entry: string;
    };
  };
}

interface State {
  contents: string[];
}

export class Show extends React.Component<RouteComponentProps<{}> & LocationStateProps, State> {
  public state: State = {
    contents: [],
  };

  @LazyInject(() => PassCli)
  private passCli: PassCli;

  public async componentDidMount(): Promise<void> {
    const {location: {state: {entry}}} = this.props;
    const contents = await this.passCli.show(entry);
    this.setState({contents});
  }

  public render(): JSX.Element {
    return (
      <div>
        {this.state.contents.map((line: string, idx: number) => <div key={idx}>{line}</div>)}
      </div>
    );
  }
}
