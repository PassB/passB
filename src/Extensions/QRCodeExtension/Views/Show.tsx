import * as React from 'react';
import {QRCode} from 'react-qr-svg';
import {RouteComponentProps} from "react-router";

import {passB} from "ConfiguredPassB";
import {PassCli} from "PassCli";

interface LocationStateProps {
  location: {
    state: {
      entry: string;
    };
  };
}

interface State {
  contents?: string[];
}

export class Show extends React.Component<RouteComponentProps<{}> & LocationStateProps, State> {
  public state: State = {};

  public async componentDidMount(): Promise<void> {
    const {location: {state: {entry}}} = this.props;
    const contents = await PassCli.show(entry);
    this.setState({contents});
  }

  public render(): JSX.Element {
    const {contents} = this.state;
    const {location: {state: {entry}}} = this.props;

    if (!contents) {
      return <div>loading....</div>;
    }
    const value = passB.getFileFormat().getPassword(contents, entry);
    if (!value) {
      return <div>file contains no password!</div>;
    }
    return (
      <QRCode
        value={value}
      />
    );
  }
}
