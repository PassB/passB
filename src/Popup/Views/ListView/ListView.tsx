import {passB} from "ConfiguredPassB";
import Menu from 'material-ui/Menu';
import MenuItem from 'material-ui/MenuItem';
import {Entry, LabeledEntries} from "PassB";
import * as React from 'react';
import Tab = browser.tabs.Tab;

interface Props {
  url: string;
  navigateTo: (newUrl: string, state: {}) => void;
}

interface State {
  entries: Entry[];
  filtered: Entry[];
}

export class ListView extends React.Component<Props, State> {
  public state: State = {
    entries: [],
    filtered: [],
  };

  public componentDidMount(): void {
    passB.getEntries()
      .then((entries: LabeledEntries) =>
        this.setState({entries: Object.values(entries)}, () => this.recalculateFilteredEntries()),
      );
  }

  public render(): JSX.Element {
    const {navigateTo} = this.props;
    const {filtered} = this.state;

    return (
      <Menu autoWidth={false} width={400} maxHeight={400}>
        {filtered && filtered.map((entry: Entry) => (
          <MenuItem
            key={entry.label}
            onClick={() => navigateTo(`entry`, {entry})}
            primaryText={entry.label.replace(/\//g, "/\u200b")}
          />
        ))}
      </Menu>
    );
  }

  private recalculateFilteredEntries(): void {
    if (!this.props.url || !this.state.entries) {
      return;
    }
    passB.getMatcher()
      .filterEntries(this.props.url || '', this.state.entries)
      .then((filtered: Entry[]) => this.setState({filtered}));
  }
}
