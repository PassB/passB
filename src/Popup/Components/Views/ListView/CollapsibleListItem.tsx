import {ListItem, ListItemText} from 'material-ui';
import {ExpandLess, ExpandMore, Folder, InsertDriveFile} from 'material-ui-icons';
import {withStyles, ClassProps, Theme} from 'material-ui/styles';
import Collapse from 'material-ui/transitions/Collapse';
import Avatar from 'material-ui/Avatar';
import * as React from 'react';
import {EntryNode} from 'PassB';
import {EntryNodeList} from './EntryNodeList';

interface Props {
  node: EntryNode;
}

interface State {
  collapsed: boolean;
}

const styles = (theme: Theme) => ({
  nested: {
    paddingLeft: theme.spacing.unit * 4,
  },
});

export const CollapsibleListItem = withStyles(styles)(
  class extends React.Component<Props & ClassProps<{ nested: Partial<React.CSSProperties> }>, State> {
    public state: State = {
      collapsed: true,
    };

    public render(): JSX.Element {
      const {node, classes} = this.props;
      const {collapsed} = this.state;

      return (
        <div>
          <ListItem
            button={true}
            key={node.fullPath}
            onClick={() => this.setState({collapsed: !collapsed})}
          >
            <Avatar>
              {node.fullPath.endsWith('/') ?
                <Folder/> :
                <InsertDriveFile/>
              }
            </Avatar>
            <ListItemText primary={node.fullPath.replace(/\//g, '/\u200b')}/>
            {collapsed ?
              <ExpandMore/> :
              <ExpandLess/>
            }
          </ListItem>
          <Collapse in={!collapsed} transitionDuration="auto" unmountOnExit={true}>
            <div className={classes.nested}>
              <EntryNodeList root={node}/>
            </div>
          </Collapse>
        </div>
      );
    }
  });
