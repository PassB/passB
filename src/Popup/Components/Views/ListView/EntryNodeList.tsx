import {Divider, List, ListItem, ListItemText} from 'material-ui';
import {Folder, InsertDriveFile} from 'material-ui-icons';
import Avatar from 'material-ui/Avatar';
import * as React from 'react';
import {withRouter, RouteComponentProps} from 'react-router';
import {LazyInject} from 'Decorators/LazyInject';
import {PassB} from 'PassB';
import {Action, EntryNode} from 'State/PassEntries/Interfaces';
import {CollapsibleListItem} from './CollapsibleListItem';

interface Props {
  root: EntryNode;
}

class UnconnectedEntryNodeList extends React.Component<Props & RouteComponentProps<{}>> {
  @LazyInject(() => PassB)
  private passB: PassB;

  public render(): JSX.Element {
    const {root, history} = this.props;

    const actionItems = root.get('actions').map((action: Action) => (
      <ListItem
        button={true}
        key={`${action.get('extension')}/${action.get('action')}`}
        onClick={() =>
          this.passB.getExtension(action.get('extension')).executeAction(
            action.get('action'),
            root.get('fullPath'),
            {navigateTo: (newUrl: string, state: {}) => history.push(newUrl, state)},
          )
        }
      >
        <ListItemText
          primary={
            browser.i18n.getMessage(this.passB.getExtension(action.get('extension')).getLabelForAction(action.get('action')))
          }
        />
      </ListItem>
    ));

    const childItems = root.get('children').map((child: EntryNode) => (
      <CollapsibleListItem
        key={child.get('fullPath')}
        CollapsedChildren={() => <UnconnectedEntryNodeList {...this.props} root={child}/>}
      >
        {[
          <Avatar key="avatar">
            {child.get('fullPath').endsWith('/') ?
              <Folder/> :
              <InsertDriveFile/>
            }
          </Avatar>,
          <ListItemText key="text" primary={child.get('fullPath').replace(/\//g, '/\u200b')}/>,
        ]}
      </CollapsibleListItem>
    ));

    return (
      <List>
        {Array.from(actionItems.values())}
        {actionItems.count() > 0 && childItems.count() > 0 && <Divider/>}
        {Array.from(childItems.values())}
      </List>
    );
  }
}

export const EntryNodeList = withRouter<Props>(UnconnectedEntryNodeList);
