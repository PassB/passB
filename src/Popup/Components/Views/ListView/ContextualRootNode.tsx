import {List} from 'immutable';
import {ListItemText} from 'material-ui';
import * as React from 'react';
import {connect} from 'react-redux';
import {createSelector} from 'reselect';
import {Matcher} from 'InjectableInterfaces/Matcher';
import {StoreContents} from 'InjectableInterfaces/State';
import {EntryNode} from 'State/PassEntries/Interfaces';
import {getCombinedPassEntries} from 'State/PassEntries/Selectors';
import {deepFilterEntryNodes, flattenEntryNode} from './entryNodeMethods';
import {CollapsibleListItem} from './CollapsibleListItem';
import {EntryNodeList} from './EntryNodeList';

interface ContextualProps {
  matcher: Matcher<{}>;
  url: string;
}

export const contextualRootNodeSelector = createSelector(
  getCombinedPassEntries,
  (_: {}, props: ContextualProps) => ({matcher: props.matcher, url: props.url}),
  (rootNode: EntryNode, {matcher, url}: ContextualProps): EntryNode | undefined =>
    deepFilterEntryNodes(
      rootNode,
      matcher.filterEntries(url, flattenEntryNode(rootNode)),
    ).set('actions', List()),
);

export const ContextualRootNodeComponent = ({root}: { root: EntryNode }) => {
  if (root.get('children').count() === 0) {
    return null; // tslint:disable-line:no-null-keyword
  } else {
    return (
      <CollapsibleListItem
        className="contextualRoot"
        CollapsedChildren={<EntryNodeList root={root}/>}
        initiallyExpanded={true}
      >
        <ListItemText
          primary="Contextual"
        />
      </CollapsibleListItem>
    );
  }
};

export const mapStateToProps = (state: StoreContents, props: ContextualProps) => ({
  root: contextualRootNodeSelector(state, props),
});

export const ContextualRootNode = connect(mapStateToProps)(ContextualRootNodeComponent);
