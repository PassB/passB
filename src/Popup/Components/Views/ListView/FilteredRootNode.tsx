import {ListItemText} from 'material-ui';
import * as React from 'react';
import {connect} from 'react-redux';
import {createSelector} from 'reselect';
import {StoreContents} from 'InjectableInterfaces/State';
import {EntryNode} from 'State/PassEntries/Interfaces';
import {getCombinedPassEntries} from 'State/PassEntries/Selectors';
import {deepFilterEntryNodes, flattenEntryNode} from './entryNodeMethods';
import {CollapsibleListItem} from './CollapsibleListItem';
import {EntryNodeList} from './EntryNodeList';

interface FilteredProps {
  filter: string;
}

export const filteredRootNodeSelector = createSelector(
  getCombinedPassEntries,
  (_: {}, props: FilteredProps) => ({filter: props.filter}),
  (rootNode: EntryNode, {filter}: FilteredProps): EntryNode | undefined => {
    if (!filter) {
      return rootNode;
    }
    return deepFilterEntryNodes(
      rootNode,
      flattenEntryNode(rootNode)
        .filter((node: EntryNode) => node.get('fullPath').includes(filter)),
    );
  },
);

export const FilteredRootNodeComponent = ({root, filter}: { root: EntryNode; filter: string }) => (
  <CollapsibleListItem
    className="allEntriesRoot"
    CollapsedChildren={<EntryNodeList root={root}/>}
    initiallyExpanded={!!filter}
  >
    <ListItemText
      primary={browser.i18n.getMessage('popup_node_all_items')}
    />
  </CollapsibleListItem>
);

export const mapStateToProps = (state: StoreContents, props: FilteredProps) => ({
  root: filteredRootNodeSelector(state, props),
});

export const FilteredRootNode = connect(mapStateToProps)(FilteredRootNodeComponent);
