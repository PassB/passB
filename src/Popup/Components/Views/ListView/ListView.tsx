import {OrderedMap} from 'immutable';
import debounce = require('lodash.debounce');
import {List, ListItem, ListItemText, TextField} from 'material-ui';
import {Sync} from 'material-ui-icons';
import {withStyles, WithStyles} from 'material-ui/styles';
import * as React from 'react';
import {connect} from 'react-redux';
import {compose} from 'recompose';
import {LazyInject} from 'Decorators/LazyInject';
import {PassB} from 'PassB';
import {EntryNode} from 'State/PassEntries/Interfaces';
import {getCombinedPassEntries} from 'State/PassEntries/Selectors';
import {StoreContents} from 'State/State';
import {CollapsibleListItem} from './CollapsibleListItem';
import {EntryNodeList} from './EntryNodeList';

interface Props {
  url: string;
}

interface ConnectedProps {
  rootNode: EntryNode;
}

interface State {
  filteredRootNode?: EntryNode;
  contextualRootNode?: EntryNode;
}

const styles = {
  centered: {
    width: '100%',
    textAlign: 'center',
  },
  noTopPadding: {
    paddingTop: '0px',
  },
};

function flattenEntryNode(currentNode: EntryNode, flattened: EntryNode[] = []): EntryNode[] {
  flattened.push(currentNode);
  currentNode.get('children').forEach((item: EntryNode) => flattenEntryNode(item, flattened));

  return flattened;
}

function deepFilterEntryNodes(node: EntryNode, filteredNodes: EntryNode[]): EntryNode {
  const nodeFullNames = filteredNodes.map((filtered: EntryNode) => filtered.get('fullPath'));
  return node.update(
    'children',
    (origChildren: OrderedMap<string, EntryNode>) =>
      origChildren
        .map((child: EntryNode) => deepFilterEntryNodes(child, filteredNodes))
        .filter((child: EntryNode) => nodeFullNames.includes(child.get('fullPath')) || child.get('children').count() > 0),
  );
}

type InnerProps = Props & ConnectedProps & WithStyles<keyof typeof styles>;

class ListViewComponent extends React.Component<InnerProps, State> {
  public state: State = {};

  @LazyInject(() => PassB)
  private passB: PassB;

  public constructor(props: InnerProps) {
    super(props);

    this.filterNodes = debounce(this.filterNodes.bind(this), 50);
  }

  public componentDidMount(): void {
    this.recalculateContextualEntries(this.props);
  }

  public componentWillReceiveProps(newProps: InnerProps): void {
    if (newProps.url !== this.props.url || newProps.rootNode !== this.props.rootNode) {
      this.recalculateContextualEntries(newProps);
    }
  }

  public render(): JSX.Element {
    const {classes, rootNode} = this.props;
    const {contextualRootNode, filteredRootNode} = this.state;

    return (
      <List>
        <ListItem className={classes.noTopPadding}>
          <TextField
            id="search"
            label={browser.i18n.getMessage('popup_search')}
            type="search"
            fullWidth={true}
            margin="none"
            onChange={(e: React.ChangeEvent<HTMLInputElement>) => this.filterNodes(e.target.value)}
            inputProps={{autoFocus: true, tabIndex: 0}}
          />
        </ListItem>
        {rootNode ?
          <CollapsibleListItem
            key="allEntriesRoot"
            CollapsedChildren={() => <EntryNodeList root={filteredRootNode || rootNode}/>}
            initiallyExpanded={typeof filteredRootNode !== 'undefined'}
          >
            <ListItemText
              primary="All Items"
            />
          </CollapsibleListItem> :
          <div className={classes.centered}><Sync/></div>
        }
        {contextualRootNode &&
        <CollapsibleListItem
          key="contextualRoot"
          CollapsedChildren={() => <EntryNodeList root={contextualRootNode}/>}
          initiallyExpanded={true}
        >
          <ListItemText
            primary="Contextual"
          />
        </CollapsibleListItem>
        }
      </List>
    );
  }

  private filterNodes(filter: string): void {
    const {rootNode} = this.props;
    const flattened: EntryNode[] = flattenEntryNode(rootNode);
    const filteredNodes = flattened.filter((node: EntryNode) => node.get('fullPath').includes(filter));

    this.setState({filteredRootNode: deepFilterEntryNodes(rootNode, filteredNodes)});

  }

  private async recalculateContextualEntries(props: InnerProps): Promise<void> {
    const {rootNode, url} = props;
    if (!url || !rootNode) {
      return;
    }

    const matcher = await this.passB.getMatcher();
    const filteredNodes = await matcher.filterEntries(url, flattenEntryNode(rootNode));
    this.setState({contextualRootNode: deepFilterEntryNodes(rootNode, filteredNodes)});
  }
}

export const ListView = compose<InnerProps, Props>(
  withStyles<keyof typeof styles>(styles),
  connect<ConnectedProps>((state: StoreContents) => ({
    rootNode: getCombinedPassEntries(state),
  })),
)(ListViewComponent);
