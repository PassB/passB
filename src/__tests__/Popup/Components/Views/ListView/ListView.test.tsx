import {shallow, ShallowWrapper} from 'enzyme';
import * as React from 'react';
import {container, Symbols} from 'Container';
import {ListView} from 'Popup/Components/Views/ListView';
import {ContextualRootNode} from 'Popup/Components/Views/ListView/ContextualRootNode';
import {FilteredRootNode} from 'Popup/Components/Views/ListView/FilteredRootNode';

const searchQuery = '.search';
const rootQuery = FilteredRootNode;
const contextualQuery = ContextualRootNode;

describe('ListView Component', () => {

  beforeAll(() => {
    container.rebind(Symbols.PassB).toConstantValue({getMatcher: () => 'mockedMatcher'});
  });

  it('renders', () => {
    const rendered = shallow(<ListView url=""/>).dive();
    expect(rendered).toMatchSnapshot();
  });

  describe('search field', () => {
    it('is present', () => {
      const rendered = shallow(<ListView url=""/>).dive();
      expect(rendered.find(searchQuery).length).toBe(1);
    });

    it('is a controlled input field', async () => {
      const rendered = shallow(<ListView url=""/>).dive();
      let search = rendered.find(searchQuery);

      expect(search.prop('value')).not.toBeUndefined();
      expect(search.prop('value')).not.toBe('newValue');
      (search as ShallowWrapper<{ onChange: Function }>).prop('onChange')({target: {value: 'newValue'}});

      rendered.update();
      search = rendered.find(searchQuery) as ShallowWrapper;
      expect(search.prop('value')).toBe('newValue');
    });
  });

  describe('Root Node', () => {
    it('is present', () => {
      const rendered = shallow(<ListView url=""/>).dive();
      expect(rendered.find(rootQuery).length).toBe(1);
    });

    it('receives values from search field', () => {
      const rendered = shallow(<ListView url=""/>).dive();
      (rendered.find(searchQuery) as ShallowWrapper<{ onChange: Function }>).prop('onChange')({target: {value: 'newValue'}});

      rendered.update();

      expect(rendered.find(rootQuery).prop('filter')).toBe('newValue');
    });
  });

  describe('Contextual Node', () => {
    it('is not present if the url is empty', () => {
      const rendered = shallow(<ListView url=""/>).dive();
      expect(rendered.find(contextualQuery).length).toBe(0);
    });

    it('is present if the url is not empty', () => {
      const rendered = shallow(<ListView url="test"/>).dive();
      expect(rendered.find(contextualQuery).length).toBe(1);
    });

    it('is passed the url from the ListView prop', () => {
      const rendered = shallow(<ListView url="testTest"/>).dive();
      expect(rendered.find(contextualQuery).prop('url')).toBe('testTest');
    });

    it('is passed the the current Matcher', () => {
      const rendered = shallow(<ListView url="testTest"/>).dive();
      expect(rendered.find(contextualQuery).prop('matcher')).toBe('mockedMatcher');
    });
  });
});
