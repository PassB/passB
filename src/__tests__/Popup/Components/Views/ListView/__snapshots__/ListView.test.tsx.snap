// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListView Component renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListViewComponent
    classes={
      Object {
        "noTopPadding": "ListViewComponent-noTopPadding-1",
      }
    }
    url=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <withStyles(ListItem)
          className="ListViewComponent-noTopPadding-1"
        >
          <TextField
            className="search"
            fullWidth={true}
            inputProps={
              Object {
                "autoFocus": true,
                "tabIndex": 0,
              }
            }
            label="popup_search_translated"
            margin="none"
            onChange={[Function]}
            required={false}
            select={false}
            type="search"
            value=""
          />
        </withStyles(ListItem)>,
        <Connect(Component)
          filter=""
        />,
        <Connect(Component)
          matcher="mockedMatcher"
          url=""
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <TextField
            className="search"
            fullWidth={true}
            inputProps={
              Object {
                "autoFocus": true,
                "tabIndex": 0,
              }
            }
            label="popup_search_translated"
            margin="none"
            onChange={[Function]}
            required={false}
            select={false}
            type="search"
            value=""
          />,
          "className": "ListViewComponent-noTopPadding-1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "className": "search",
            "fullWidth": true,
            "inputProps": Object {
              "autoFocus": true,
              "tabIndex": 0,
            },
            "label": "popup_search_translated",
            "margin": "none",
            "onChange": [Function],
            "required": false,
            "select": false,
            "type": "search",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "filter": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "matcher": "mockedMatcher",
          "url": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <withStyles(ListItem)
            className="ListViewComponent-noTopPadding-1"
          >
            <TextField
              className="search"
              fullWidth={true}
              inputProps={
                Object {
                  "autoFocus": true,
                  "tabIndex": 0,
                }
              }
              label="popup_search_translated"
              margin="none"
              onChange={[Function]}
              required={false}
              select={false}
              type="search"
              value=""
            />
          </withStyles(ListItem)>,
          <Connect(Component)
            filter=""
          />,
          <Connect(Component)
            matcher="mockedMatcher"
            url=""
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <TextField
              className="search"
              fullWidth={true}
              inputProps={
                Object {
                  "autoFocus": true,
                  "tabIndex": 0,
                }
              }
              label="popup_search_translated"
              margin="none"
              onChange={[Function]}
              required={false}
              select={false}
              type="search"
              value=""
            />,
            "className": "ListViewComponent-noTopPadding-1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "className": "search",
              "fullWidth": true,
              "inputProps": Object {
                "autoFocus": true,
                "tabIndex": 0,
              },
              "label": "popup_search_translated",
              "margin": "none",
              "onChange": [Function],
              "required": false,
              "select": false,
              "type": "search",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "filter": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "matcher": "mockedMatcher",
            "url": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
